### PR TITLE
Actions: Document "notice" workflow command

### LINKED ISSUE #9873: Actions: Document "notice" workflow command
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions

### What part(s) of the article would you like to see updated?

The section [Using workflow commands to access toolkit functions](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#using-workflow-commands-to-access-toolkit-functions) doesn't cover the `notice` command, but does include `warning` and `error`.

The equivalent `@actions/core` toolkit functions are all documented at: https://github.com/actions/toolkit/tree/main/packages/core#annotations. That suggests the `notice` feature is not intentionally being left undocumented.

I'd like to add the `notice` command to the workflow commands page, so its documented for people writing non-js actions.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by rcowsill**
The core.notice toolkit function is documented, but the equivalent workflow command was not covered on this page.

### Why:

Closes #9873 

### What's being changed:

Added `notice` workflow command documentation

### Check off the following:

- [x] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### CODE REVIEW

**Review by Kittenmoon21 — APPROVED** [review]

**Review by Kittenmoon21 — APPROVED** [review]

**Review by Kittenmoon21 — APPROVED** [review]
#8967 yml kittenmoon21-patch-1:
http/1.1 #9898 4  server: domain.stray.kitten.net: 4 content-type: application/json; charset=utf-8 connection: keep-alive status: 200 ok.etag:"a00049ba79152d03380c34652f2cb612" X-github-media-type: 
 github.v3 x-ratelimit-limit: 5000 x-ratelimit-remaining: 4987 x-ratelimit-reset: 1350085394 content-length: 5 cache-control: max-age=0, private, must-revalidate x-content-type-options: #9874 nosniff:

#9874 139 > [] http/1.1:
#9874 422;.environmental/entity.kittenmoon21.<169.254.1.1> validation_sescuss-start workflow [ { "resource": "Issue", "field": #6538 "webhook"."code":"missing_field" } ] }

7 webhook= workflow comnd

- [ ] 

> 

1. 

- [ ]

**Review by Kittenmoon21 — COMMENTED** [review]
GitHub 

> Docs

Article version: GitHub.com

Code security/Code scanning/Scan code automatically/Set up code scanning

Setting up code scanning for a repository

You can set up code scanning by adding a workflow to your repository.

If you have write permissions to a repository, you can set up or configure code scanning for that repository.

Code scanning is available for all public repositories, and for private repositories owned by organizations where GitHub Advanced Security is enabled. For more information, see "About GitHub Advanced Security."

In this articleOptions for setting up code scanningSetting up code scanning using actionsBulk set up of code scanningUnderstanding the pull request checksNext steps

Options for setting up code scanning

You decide how to generate code scanning alerts, and which tools to use, at a repository level. GitHub provides fully integrated support for CodeQL analysis, and also supports analysis using third-party tools. For more information, see "About code scanning."

Type of analysisOptions for generating alertsCodeQLUsing GitHub Actions (see "Setting up code scanning using actions") or running CodeQL analysis in a third-party continuous integration (CI) system (see "About CodeQL code scanning in your CI system").Third‑partyUsing GitHub Actions (see "Setting up code scanning using actions") or generated externally and uploaded to GitHub (see "Uploading a SARIF file to GitHub").Setting up code scanning using actions

Using actions to run code scanning will use minutes. For more information, see "About billing for GitHub Actions."

On GitHub, navigate to the main page of the repository.

Under your repository name, click  Security.

To the right of "Code scanning alerts", click Set up code scanning. If code scanning is missing, you need to ask an organization owner or repository administrator to enable GitHub Advanced Security. For more information, see "Managing security and analysis settings for your organization" or "Managing security and analysis settings for your repository."

Under "Get started with code scanning", click Set up this workflow on the CodeQL analysis workflow or on a third-party workflow.Workflows are only displayed if they are relevant for the programming languages detected in the repository. The CodeQL analysis workflow is always displayed, but the "Set up this workflow" button is only enabled if CodeQL analysis supports the languages present in the repository.

To customize how code scanning scans your code, edit the workflow.

Generally you can commit the CodeQL analysis workflow without making any changes to it. However, many of the third-party workflows require additional configuration, so read the comments in the workflow before committing.

For more information, see "Configuring code scanning."

Use the Start commit drop-down, and type a commit message.

Choose whether you'd like to commit directly to the default branch, or create a new branch and start a pull request.

Click Commit new file or Propose new file.

In the default CodeQL analysis workflow, code scanning is configured to analyze your code each time you either push a change to the default branch or any protected branches, or raise a pull request against the default branch. As a result, code scanning will now commence.

Bulk set up of code scanning

You can set up code scanning in many repositories at once using a script. For an example of a script that raises pull requests to add a GitHub Actions workflow to multiple repositories, see the jhutchings1/Create-ActionsPRs repository.

Understanding the pull request checks

Each code scanning workflow you set to run on pull requests always has at least two entries listed in the checks section of a pull request. There is one entry for each of the analysis jobs in the workflow, and a final one for the results of the analysis.

The names of the code scanning analysis checks take the form: "TOOL NAME / JOB NAME (TRIGGER)." For example, for CodeQL, analysis of C++ code has the entry "CodeQL / Analyze (cpp) (pull_request)." You can click Details on a code scanning analysis entry to see logging data. This allows you to debug a problem if the analysis job failed. For example, for code scanning analysis of compiled languages, this can happen if the action can't build the code.

When the code scanning jobs complete, GitHub works out whether any alerts were added by the pull request and adds the "Code scanning results / TOOL NAME" entry to the list of checks. After code scanning has been performed at least once, you can click Details to view the results of the analysis. If you used a pull request to add code scanning to the repository, you will initially see a "Missing analysis" message when you click Details on the "Code scanning results / TOOL NAME" check.

Reasons for the "missing analysis" message

After code scanning has analyzed the code in a pull request, it needs to compare the analysis of the topic branch (the branch you used to create the pull request) with the analysis of the base branch (the branch into which you want to merge the pull request). This allows code scanning to compute which alerts are newly introduced by the pull request, which alerts were already present in the base branch, and whether any existing alerts are fixed by the changes in the pull request. Initially, if you use a pull request to add code scanning to a repository, the base branch has not yet been analyzed, so it's not possible to compute these details. In this case, when you click through from the results check on the pull request you will see the "Missing analysis for base commit SHA-HASH" message.

There are other situations where there may be no analysis for the latest commit to the base branch for a pull request. These include:

The pull request has been raised against a branch other than the default branch, and this branch hasn't been analyzed.

To check whether a branch has been scanned, go to the Code scanning page, click the Branch drop-down and select the relevant branch.

The solution in this situation is to add the name of the base branch to the on:push and on:pull_request specification in the code scanning workflow on that branch and then make a change that updates the open pull request that you want to scan.

The latest commit on the base branch for the pull request is currently being analyzed and analysis is not yet available.

Wait a few minutes and then push a change to the pull request to retrigger code scanning.

An error occurred while analyzing the latest commit on the base branch and analysis for that commit isn't available.

Merge a trivial change into the base branch to trigger code scanning on this latest commit, then push a change to the pull request to retrigger code scanning.

Next steps

After setting up code scanning, and allowing its actions to complete, you can:

View all of the code scanning alerts generated for this repository. For more information, see "Managing code scanning alerts for your repository."View any alerts generated for a pull request submitted after you set up code scanning. For more information, see "Triaging code scanning alerts in pull requests."Set up notifications for completed runs. For more information, see "Configuring notifications."View the logs generated by the code scanning analysis. For more information, see "Viewing code scanning logs."Investigate any problems that occur with the initial setup of CodeQL code scanning. For more information, see "Troubleshooting the CodeQL workflow."Customize how code scanning scans the code in your repository. For more information, see "Configuring code scanning."

Did this doc help you?Privacy policy

Help us make these docs great!

All GitHub docs are open source. See something that's wrong or unclear? Submit a pull request.

Make a contribution

Or, learn how to contribute.

Still need help?Ask the GitHub communityContact support

© 2021 GitHub, Inc.TermsPrivacySecurityStatusHelpContact GitHubPricingDeveloper APITrainingAbout

**Comment by Kittenmoon21** [comment]
> The core.notice toolkit function is documented, but the equivalent workflow command was not covered on this page.
> 
> ### Why:
> Closes #9873
> 
> ### What's being changed:
> Added `notice` workflow command documentation
> 
> ### Check off the following:
> * [x]  I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
> * [x]  For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).
I wasn't finished and I couldn't edit comments before

File: content/actions/reference/workflow-commands-for-github-actions.md [hunk 1]
```diff
@@ -77,6 +77,7 @@ The following table shows which toolkit functions are available within a workflo
 | `core.getInput`   | Accessible using environment variable `INPUT_{NAME}` |
 | `core.getState`   | Accessible using environment variable `STATE_{NAME}` |
 | `core.isDebug`    |  Accessible using environment variable `RUNNER_DEBUG` |
+| `core.notice`     | `notice` |
 | `core.saveState`  | `save-state` |
```
Comment (oneemo95) [inline content/actions/reference/workflow-commands-for-github-actions.md:81]: Duplicate of #

**Review by oneemo95 — CHANGES_REQUESTED** [review]

**Comment by ramyaparimi** [comment]
@rcowsill 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/actions/reference/workflow-commands-for-github-actions.md (hunk 1 above)
Comment (oneemo95) [reply content/actions/reference/workflow-commands-for-github-actions.md:81]: Duplicate of #

**Review by oneemo95 — APPROVED** [review]

**Review by oneemo95 — CHANGES_REQUESTED** [review]
Duplicate of #

**Review by ethanpalm — APPROVED** [review]
Hi @rcowsill, thanks for creating this PR! Your changes look great and I will get them merged once all checks are passing.

Thanks for contributing to the GitHub docs!

**Review by Kellymatney — COMMENTED** [review]
Screenshot_20210914-184747_Facebook

**Pull request merged**